### PR TITLE
[WIP] Implement cursor-shape-v1 protocol

### PR DESCRIPTION
**Pull request by heroin-moose**
This protocol allows Wayland clients to request a buffer for a cursor shape from a compositor.

Tested with foot.

### CODE REVIEW

File: src/input/cursor.c [hunk 1]
```diff
@@ -1225,6 +1247,17 @@ cursor_init(struct seat *seat)
...
+
+	struct wlr_cursor_shape_manager_v1 *cursor_shape_manager =
+		wlr_cursor_shape_manager_v1_create(seat->server->wl_display, 1);
+	if (!cursor_shape_manager) {
+		wlr_log(WLR_ERROR, "unable to create cursor_shape interface");
+		exit(EXIT_FAILURE);
+	}
+	seat->request_set_shape.notify = request_set_shape_notify;
+	wl_signal_add(&cursor_shape_manager->events.request_set_shape,
+		&seat->request_set_shape);
```
Comment (Consolatis) [inline src/input/cursor.c:1259]: Signal handler should be cleaned up in `cursor_finish()`

File: src/input/cursor.c [hunk 2]
```diff
@@ -158,6 +161,32 @@ request_cursor_notify(struct wl_listener *listener, void *data)
...
+	struct wlr_seat_client *focused_client = seat->seat->pointer_state.focused_client;
+
+	/* Prevent setting a cursor image when moving or resizing */
+	if (seat->server->input_mode != LAB_INPUT_STATE_PASSTHROUGH) {
+		return;
+	}
+
+	/*
+	 * This can be sent by any client, so we check to make sure this one is
+	 * actually has pointer focus first.
```
Comment (Consolatis) [inline src/input/cursor.c:178]: Nitpick: spelling, `s/ is$//`

Also in the source of the comment.

**Comment by Consolatis** [comment]
Have tested, other than the 3 minor things above this LGTM.

File: src/input/cursor.c (hunk 2 above)
Comment (heroin-moose) [reply src/input/cursor.c]: Fixed. It's a copypasta, so the one above has the same typo, fixed in another commit.

File: src/input/cursor.c [hunk 3]
```diff
@@ -158,6 +161,33 @@ request_cursor_notify(struct wl_listener *listener, void *data)
...
+	 */
+	if (event->seat_client != focused_client) {
+		wlr_log(WLR_INFO, "seat client %p != focused client %p",
+			event->seat_client, focused_client);
+		return;
+	}
+
+	wlr_log(WLR_INFO, "set xcursor to shape %s", wlr_cursor_shape_v1_name(event->shape));
```
[the remark this answers is not in the document]
Comment (heroin-moose) [reply src/input/cursor.c]: Makes sense, though I think shape_name is better.

**Comment by Consolatis** [comment]
Thanks :)

**Pull request merged**